skimage/measure/fit.py

```diff
@@ -515,7 +528,11 @@ def estimate(self, data):
         C1 = np.array([[0., 0., 2.], [0., -1., 0.], [2., 0., 0.]])
 
         # Reduced scatter matrix [eqn. 29]
-        M = np.linalg.inv(C1).dot(S1 - np.dot(S2, np.linalg.inv(S3)).dot(S2.T))
+        try:
+            M = np.linalg.inv(C1).dot(
+                S1 - np.dot(S2, np.linalg.inv(S3)).dot(S2.T))
+        except:  # LinAlgError: Singular matrix
```

Comment:
Please, be more strict on the exception you catch. Otherwise, you'll catch everything and errors can't be detected.